File: backend/app/models/cancha.py
from sqlalchemy.orm import relationship
from sqlalchemy import Column, Integer, String, Boolean
from app.db.base_class import Base

class Cancha(Base):

    id = Column(Integer, primary_key=True, index=True)
    tipo = Column(String, index=True)
    ubicacion = Column(String, index=True)
    fotografia = Column(String, index=True)
    estado_disponibilidad = Column(Boolean, default=True)

    reservations = relationship("Reservation", back_populates="cancha")
    horarios = relationship("Horario", back_populates="cancha")
    
    

File: backend/app/models/cancha.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Cancha extends Model {
    static associate(models) {
      Cancha.hasMany(models.Reservation, {
        foreignKey: 'canchaId',
        as: 'reservations',
      });
      Cancha.hasMany(models.Horario, {
        foreignKey: 'canchaId',
        as: 'horarios',
      });
    }
  }
  Cancha.init({
    nombre: DataTypes.STRING,
    tipo: DataTypes.STRING,
    ubicacion: DataTypes.STRING,
    fotografia: DataTypes.STRING,
    estado_disponibilidad: DataTypes.BOOLEAN
  }, {
    sequelize,
    modelName: 'Cancha',
  });
  return Cancha;
};